tests/test_generator/test_generator.py

```diff
@@ -0,0 +1,269 @@
+import filecmp
+import os
+import pathlib
+from glob import glob
+from logging import DEBUG, INFO
+from os import environ, listdir
+from os import path as osp
+
+import pytest
+
+from smartsim import Experiment
+from smartsim._core.generation.generator import Generator
+from smartsim.entity import Application, Ensemble
+from smartsim.launchable import Job
+from smartsim.settings.launchSettings import LaunchSettings
+
+# TODO test ensemble copy, config, symlink when ensemble.attach_generator_files added
```

Comment:
The Ensemble class does not yet have a method named `attach_generator_files` - are suggesting that I add this to the Ensemble class? Or oop, will a user purely expand files via Ensemble.files param?

Reply:
OH! That's what you meant by this comment!

I guess I hadn't considered if we were adding an `attach_generator_files` method to `Ensemble`s. If we are, yeah, the tests should definitely be added when that work is done, and not in this PR. I would still remove the TODO just because I think the unit tests should just check that the state of the ensemble is changed, and the unit tests you have for generation are already sufficient!